### ISSUE TITLE
[Improvement] Destructuring improvement (computed object property names)

### ISSUE THREAD

**Issue by da411d** [javascript-obfuscator/javascript-obfuscator#720]
As continuation to #688 i have one more idea to improve destructuring.

Example code:
```
const { getSecret, getSuperSecret } = require("./secret");

const secret = getSecret();
const superSecret = getSuperSecret();

console.log(secret, superSecret);
```

First improvement was to [reassign it to other variable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assigning_to_new_variable_names)
```
const { getSecret: p, getSuperSecret: q } = require("./secret");

const secret = p();
const superSecret = q();

console.log(secret, superSecret);
```

Now i propose to use _computed object property names_:
It works in next way:
```
const keyName = "user_name";
const obj = {
  [keyName]: "foo bar baz boo",
};
// {user_name: foo bar baz boo"}
```

Also it [works fine with destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Computed_object_property_names_and_destructuring):
```
const e = "getSecret"; // string will be obfuscated with stringArray or something else maybe
const f = "getSuperSecret";
const { [e]: p, [f]: q } = require("./secret");

const secret = p();
const superSecret = q();

console.log(secret, superSecret);
```

**Comment by sanex3339** [comment]
Good idea but i need to think about how to do this because right now same mechanics could be done with `controlFlowFlattening` (it moves strings to the object), BUT it runs BEFORE `converting` transformers, so transformer that converts `foo.bar` to the `foo['bar']` runs after `controlFlowFlattening` stage. 
But if i move `controlFlowFlattening` stage after `converting` stage, these transformers are ignore all code added by `controlFlowFlattening` stage, especially objects.

So this is not an easy dependencies loop and i need to think how to fix it.
I've already done some refactoring related to transformers structure and order here: https://github.com/javascript-obfuscator/javascript-obfuscator/pull/721

But this PR does not fix the problem described above.
